src/test/java/pkmncore/PokemonFinderTest.java

```diff
@@ -0,0 +1,45 @@
+package pkmncore;
+
+import com.google.gson.JsonObject;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class PokemonFinderTest {
+
+    private SearchEngine search = new SearchFake();
+    private PokemonFinder pokemonFinder = new PokemonFinder(search);
+
+    @Test
+    public void findsPokemonName() {
+        JsonObject data = search.findByName("pikachu");
```

Comment:
I think you don't necessarily have to explicitly call the searchFake to get the expected object back before the assertion. 

Pokemon pokemon = pokemonFinder.findName(data);
You could do assertEquals("pikachu", pokemon.getName()); for example.